diff --git a/drivers/usb/phy/phy-mxs-usb.c b/drivers/usb/phy/phy-mxs-usb.c
index dcb447db5..c7a5a984b 100644
--- a/drivers/usb/phy/phy-mxs-usb.c
+++ b/drivers/usb/phy/phy-mxs-usb.c
@@ -828,6 +828,10 @@ static enum usb_charger_type mxs_phy_charger_detect(struct usb_phy *phy)
 	void __iomem *base = phy->io_priv;
 	enum usb_charger_type chgr_type = UNKNOWN_TYPE;
 
+	dev_info(mxs_phy->phy.dev,
+		 "refusing to perform mxs_phy_charger_detect\n");
+	return UNKNOWN_TYPE;
+
 	if (!regmap)
 		return UNKNOWN_TYPE;
 
