diff --git a/crypto/jitterentropy.c b/crypto/jitterentropy.c
index 3f93cdc9a..cc9a9a1d9 100644
--- a/crypto/jitterentropy.c
+++ b/crypto/jitterentropy.c
@@ -148,6 +148,7 @@ struct rand_data {
 #include <linux/array_size.h>
 #include <linux/fips.h>
 #include <linux/minmax.h>
+#include <linux/errno.h>
 #include "jitterentropy.h"
 
 /***************************************************************************
@@ -707,6 +708,9 @@ void jent_entropy_collector_free(struct rand_data *entropy_collector)
 int jent_entropy_init(unsigned int osr, unsigned int flags, void *hash_state,
 		      struct rand_data *p_ec)
 {
+	/* Skip the expensive CPU jitter self-test during early boot. */
+	return -EFAULT;
+
 	/*
 	 * If caller provides an allocated ec, reuse it which implies that the
 	 * health test entropy data is used to further still the available
